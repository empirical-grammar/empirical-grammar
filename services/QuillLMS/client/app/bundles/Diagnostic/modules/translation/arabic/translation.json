{
    "buttons": {
        "begin": "Begin / ابدأ",
        "continue": "Continue / استمر",
        "next": "Next / استمر",
        "resume": "Resume / استمر",
        "submit": "Submit / إرسال"
    },
    "intro": {
        "header": "نشاط تحديد المستوى في Quill",
        "text": "أنت على وشك الإجابة على 22 سؤالًا حول كتابة جمل. قد تكون بعض الأسئلة حول أشياء لم تتعلمها بعد – لا بأس! فقط أجب بأفضل طريقة ممكنة. لا تنس قراءة التعليمات بعناية لكل سؤال! بمجرد الانتهاء، ستقوم Quill بإنشاء خطة تعليمية لك فقط."
    },
    "ELL Starter Fill in the Blank": {
        "header": "املأ الفراغ",
        "text": "في هذا القسم، سيكون لكل جملة فراغ واحد أو اثنان أو ثلاثة. ستملأ كل فراغ بأحد الكلمات من القائمة، أو ستترك الفراغ دون ملء إذا لم تكن هناك حاجة إلى كلمة. اكتب الكلمات التي تختارها في المربعات. تذكر أن تتأكد من إملائك! هيا بنا نبدأ!"
    },
    "ELL Starter Halfway": {
        "header": "ت في منتصف الطريق!",
        "text": "ستمر في العمل الرائع!"
    },
    "ELL Starter Questions": {
        "header": "الأسئلة",
        "text": "في هذا القسم، سوف تكتب الأسئلة. تذكر أن الأسئلة لها علامة استفهام (؟) في النهاية. ستنجح في هذا!"
    },
    "ELL Starter Sentence Combining": {
        "header": "جمع الجمل",
        "text": "لقد انتهيت تقريبًا! بقي 3 أسئلة فقط. في هذا القسم، سوف تجمع الجمل في جملة واحدة. ستحصل على كلمة ربط لتستخدمها. تذكر أن الجملة لها نقطة واحدة فقط. ستنجح في هذا!"
    },
    "End: You Did it!": {
        "header": "لقد نجحت!",
        "text": "انتهى الأمر! لقد انتهيت للتو من نشاط تحديد المستوى الخاص ب Quill. سيتم إرسال النتائج إلى معلمك، وسنبدأ في إنشاء خطة التعلم الخاصة بك."
    },
    "completedDiagnostic": {
        "header": "لقد أكملت نشاط تحديد المستوى الخاص ب Quill",
        "text": "يتم حفظ نتائجك الآن. سيتم إعادة توجيهك تلقائيًا بمجرد حفظها."
    },
    "instructions": {
        "Fill in the blank with one of the choices.": "املأ الفراغ بأحد الخيارات.",
        "Fill in each blank with one of the choices.": "املأ كل فراغ بأحد الخيارات.",
        "Fill in the blanks with one word each.": "املأ الفراغات بكلمة واحدة لكل منها.",
        "Fill in the blanks with one word each. You can leave one empty.": "املأ الفراغات بكلمة واحدة. يمكنك ترك أحد الفراغات دون ملء.",
        "Use the joining word to combine the sentences into one sentence.": "استخدم كلمة الربط لدمج الجمل في جملة واحدة."
    },
    "cues": {
        "choose one": "choose one / اختر واحدًا",
        "choose three": "choose three / اختر ثلاثة",
        "choose two or three" : "choose two or three / اختر اثنين أو ثلاثة",
        "joining word": "joining word / كلمة الربط"
    },
    "feedback": {
        "blank": "استخدم أحد الخيارات أدناه أو اتركه فارغًا.",
        "no blank": "استخدم أحد الخيارات أدناه."
    }
}
