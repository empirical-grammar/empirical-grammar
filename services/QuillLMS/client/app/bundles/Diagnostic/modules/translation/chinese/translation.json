{
    "buttons": {
        "begin": "Begin / 开始",
        "continue": "Continue / 继续",
        "next": "Next / 继续",
        "resume": "Resume / 继续",
        "submit": "Submit / 提交"
    },
    "intro": {
        "header": "Quill安置活动",
        "text": "你将要回答22个关于编写句子的问题。有些问题可能与你还没学过的东西有关——没关系！尽你所能地回答他们。不要忘记仔细阅读每个问题的说明！在你完成后，Quill会为你制定一份学习计划。"
    },
    "ELL Starter Fill in the Blank": {
        "header": "填空",
        "text": "在本节中，每个句子会有一个、两个或三个空格。你要用列表中的其中一个词语填写每个空格，或者如果不需要词语，则将空格留空。请在框中输入你选择的词语。记得检查你的拼写！我们开始吧！"
    },
    "ELL Starter Halfway": {
        "header": "你已经完成一半了！",
        "text": "继续努力吧！"
    },
    "ELL Starter Questions": {
        "header": "问题",
        "text": "在本节中，由你来编写问题。请记住，问题的末尾带有问号（？）。你能做到的！"
    },
    "ELL Starter Sentence Combining": {
        "header": "句子组合",
        "text": "你即将大功告成！只剩三个问题了。在本节中，你要把几个句子组合成一个句子。会给你一个连接词来使用。请记住，一个句子只有一个句号。你肯定没问题的！"
    },
    "End: You Did it!": {
        "header": "你完成了！",
        "text": "就是这样！你刚刚完成了Quill安置活动。结果将发送给你的老师，我们将开始制定你的学习计划。"
    },
    "completedDiagnostic": {
        "header": "你已经完成了Quill安置活动",
        "text": "现在正在保存你的结果。保存后，你会被自动重定向。"
    },
    "instructions": {
        "Fill in the blank with one of the choices.": "用其中一个选项填写空格。",
        "Fill in each blank with one of the choices.": "用其中一个选项填写每个空格。",
        "Fill in the blanks with one word each.": "每个空格填写一个词语。",
        "Fill in the blanks with one word each. You can leave one empty.": "每个空格填写一个词语。你可以留下一处空白",
        "Use the joining word to combine the sentences into one sentence.": "用连接词把几个句子组合成一个句子。"
    },
    "cues": {
        "choose one": "choose one / 选择一个",
        "choose three": "choose three / 选择三个",
        "choose two or three" : "choose two or three / 选择两个或三个",
        "joining word": "joining word / 连接词"
    },
    "feedback": {
        "blank": "使用下面的其中一个选项，或者留空。",
        "no blank": "使用下面的其中一个选项。"
    }
}
