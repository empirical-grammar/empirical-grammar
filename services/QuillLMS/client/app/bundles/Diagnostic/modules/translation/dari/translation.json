{
    "buttons": {
        "begin": "Begin / شروع",
        "continue": "Continue / جاری",
        "next": "Next / جاری",
        "resume": "Continue / جاری",
        "submit": "Submit / ارسال"
    },
    "intro": {
        "header": "فعالیت قرارگیری درکویل Quill",
        "text": "شمادارین 22 سوال راجواب بدهید با نوشتن جملات. بعضی جملات درباره چیزهای است که شاید شما تا هنوزیادنداشته باشید-درست است! تامیتوانید جواب درست بدهید. ازیادتان نرود که هدایت هرسوال رادرست بخوانید!که خلاص شد,کویل به شما پلان یادگرفتن ایجاد میکند."
    },
    "ELL Starter Fill in the Blank": {
        "header": "خانه خالی را پورنماید",
        "text": "دراین بخش هرجمله یک,دو یا سه خانه خالی دارد.شماپورکنید هرخانه خالی رابایکی ازکلمات درلیست. کلماتی را که انتخاب می کنید در جعبه ها تایپ کنید. به یاد داشته باشید که املای خود را بررسی کنید! بیایید شروع کنیم!"
    },
    "ELL Starter Halfway": {
        "header": "شماه درنیمه راه هستین!",
        "text": "کارعالی تان ادامه دهید!"
    },
    "ELL Starter Questions": {
        "header": "سوالات",
        "text": "دراین بخش شما سوالات را نوشته خواهید کرد. بیاد داشته باشید که سوالات علامه سوالیه دارد(؟) دراخر. شمامیتوانید!"
    },
    "ELL Starter Sentence Combining": {
        "header": "وصل کردن جمله",
        "text": "شما تقریباٌ خلاص کردین! فقط 3 سوال مانده. دراین بخش, شما جملات راوصل میکنید که یک جمله شود. کلمه وصل کردن به شماداده خواهد شد. به یاد داشته باشید که یک جمله فقط یک دوره دارد. شما این را فهمیدین!"
    },
    "End: You Did it!": {
        "header": "شما انجام دادید!",
        "text": "بس همین قدربود! شما به تازگی فعالیت جایگذاری Quill را تمام کرده اید. نتایج به معلم شما ارسال می شود ، و ما در ایجاد برنامه یادگیری شما شروع خواهیم کرد."
    },
    "completedDiagnostic": {
        "header": "شما فعالیت جایگذاری Quill را کامل کردید",
        "text": "نتایج شما اکنون ذخیره می شود. پس از ذخیره به صورت خودکار هدایت می شوید."
    },
    "instructions": {
        "Fill in each blank with one of the choices. You can leave one empty.": "در هر خانه خالی با یکی از انتخاب ها پر کنید. شما می توانید یک خانه خالی را ترک کنید.",
        "Fill in the blank with one of the choices.": "بایکی از گزینه ها خانه خالی را پورکنید",
        "Fill in each blank with one of the choices.": " بایکی از گزینه ها هرخانه خالی راپورنماید",
        "Fill in the blanks with one word each.": "هرخانه خالی رابایک کلمه پورکنید",
        "Fill in the blanks with one word each. You can leave one empty.": "هرخانه خالی رابایک کلمه پورکنید. میتوانید یکی راخالی بمانید.",
        "Use the joining word to combine the sentences into one sentence.": "برای ترکیب جملات با یک جمله از کلمه وصل کردن استفاده کنید"
    },
    "cues": {
        "type one": "type one / یک تایپ کنید",
        "type two or three": "type two or three / دو یا سه تایپ کنید",
        "type three": "type three / سه تایپ کنید",
        "choose one": "choose one / یکی راانتخاب کن",
        "choose three": "choose three / سه تا انتخاب کن",
        "choose two or three" : "choose two or three / دویا سه تا انتخاب کن",
        "joining word": "joining word / کلمه وصل یا الحاق"
    },
    "feedback": {
        "blank": "ازیکی گزینه زیراستفاده کیند یا خالی بگذارید.",
        "no blank": "ازیکی گزینه زیراستفاده کنید."
    }
}
