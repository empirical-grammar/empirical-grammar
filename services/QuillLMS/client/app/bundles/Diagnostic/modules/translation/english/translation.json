{
    "buttons": {
        "begin": "Begin",
        "continue": "Continue",
        "next": "Next",
        "resume": "Resume",
        "submit": "Submit"
    },
    "cues": {
        "choose one": "choose one",
        "choose three": "choose three",
        "choose two or three" : "choose two or three",
        "joining word": "joining word"
    }
}
