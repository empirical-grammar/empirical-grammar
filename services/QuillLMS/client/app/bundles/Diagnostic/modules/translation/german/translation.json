{
    "buttons": {
        "begin": "Begin / Anfangen",
        "continue": "Continue / Weiter",
        "next": "Next / Weiter",
        "resume": "Continue / Weiter",
        "submit": "Submit / Einreichen"
    },
    "intro": {
        "header": "Quill Platzdeckchen-Methode (Placemat)",
        "text": "Du stehst kurz davor, 22 Fragen über das Schreiben von Sätzen zu beantworten. Einige der Fragen könnten sich auf Dinge beziehen, die du noch nicht gelernt hast – das ist in Ordnung! Beantworte sie einfach, so gut du kannst. Vergiss nicht, die Anweisungen für jede Frage sorgfältig zu lesen! Sobald du fertig bist, wird Quill einen Lernplan für dich erstellen."
    },
    "ELL Starter Fill in the Blank": {
        "header": "Fülle die Lücken aus",
        "text": "Jeder Satz in diesem Abschnitt hat eine, zwei oder drei Lücken. Du kannst entweder jede Lücke mit einem der Wörter aus der Liste ausfüllen oder die Lücke leer lassen, falls kein Wort benötigt wird. Gib die Wörter, die du ausgewählt hast, in die Felder ein. Vergiss nicht, deine Rechtschreibung zu überprüfen! Lass uns loslegen!"
    },
    "ELL Starter Halfway": {
        "header": "Du bist schon halbwegs fertig!",
        "text": "Großartige Arbeit, mach weiter so!"
    },
    "ELL Starter Questions": {
        "header": "Fragen",
        "text": "In diesem Abschnitt wirst du Fragen schreiben. Denke daran, dass am Ende jeder Frage ein Fragezeichen (?) stehen muss. Du schaffst das schon!"
    },
    "ELL Starter Sentence Combining": {
        "header": "Sätze kombinieren",
        "text": "Du bist fast fertig! Es sind nur noch 3 Fragen übrig. In diesem Abschnitt wirst du mehrere Sätze zu einem einzigen Satz kombinieren. Dafür wirst du ein Verbindungswort erhalten. Denke daran, dass ein Satz nur einen Punkt hat. Du hast das hier im Griff!"
    },
    "End: You Did it!": {
        "header": "Du hast es geschafft!",
        "text": "Das war schon alles! Du hast gerade die Platzdeckchen-Methode von Quill beendet. Die Ergebnisse werden an deinen Lehrer geschickt und wir fangen mit der Erstellung deines Lernplans an."
    },
    "completedDiagnostic": {
        "header": "Du hast die Platzdeckchen-Methode von Quill abgeschlossen.",
        "text": "Deine Ergebnisse werden jetzt gespeichert. Sobald diese gespeichert sind, wirst du automatisch weitergeleitet."
    },
    "instructions": {
        "Fill in the blank with one of the choices.": "Fülle die Lücken mit einer der möglichen Optionen aus.",
        "Fill in each blank with one of the choices.": "Fülle jede Lücke mit einer der möglichen Optionen aus.",
        "Fill in the blanks with one word each.": "Fülle die Lücken mit je einem Wort aus.",
        "Fill in the blanks with one word each. You can leave one empty.": "Fülle die Lücken mit je einem Wort aus oder lasse sie leer.",
        "Use the joining word to combine the sentences into one sentence.": "Benutze das Verbindungswort, um Sätze zu einem einzigen Satz zu verbinden."
    },
    "cues": {
        "choose one": "choose one / wähle eins",
        "choose three": "choose three / wähle drei",
        "choose two or three" : "choose two or three / wähle zwei oder drei",
        "joining word": "joining word / verbindungswort"
    },
    "feedback": {
        "blank": "Verwende eine der untenstehenden Optionen oder lass sie leer.",
        "no blank": "Verwende eine der untenstehenden Optionen."
    }
}
