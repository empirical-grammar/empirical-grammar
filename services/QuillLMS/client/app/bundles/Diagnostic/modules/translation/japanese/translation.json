{
    "buttons": {
        "begin": "Begin / 開始",
        "continue": "Continue / 続行",
        "resume": "Continue / 続行",
        "submit": "Submit / 送信"
    },
    "intro": {
        "header": "Quillクラス分けアクティビティ",
        "text": "作文についての22問の質問に答えます。質問には、学習していない事柄についてのものもありますが、心配は不要です！単にベストを尽くして回答すれば結構です！必ず各質問の説明を注意深く読んでください。回答を終えたら、Quillであなたに最適な学習プランが作成されます。"
    },
    "ELL Starter Fill in the Blank": {
        "header": "空欄を埋めます",
        "text": "このセクションには、各文に1つから3つの空欄があります。各空欄にリスト内の単語から1つ選んで記入するか、単語を記入する必要がない場合は記入せずそのままにします。選んだ単語を四角の中に記入します。綴りのチェックを忘れないでください！さあ、始めましょう！"
    },
    "ELL Starter Halfway": {
        "header": "半分終わりました！",
        "text": "その調子で頑張りましょう！"
    },
    "ELL Starter Questions": {
        "header": "質問",
        "text": "このセクションでは、あなたが質問を書きます。質問文の文末に忘れずに疑問符（？）をつけてください。あなたならできますよ！"
    },
    "ELL Starter Sentence Combining": {
        "header": "文章の結合",
        "text": "あと少しで終了です！残りわずか3問です。このセクションでは、複数の文章を1つの文章に結合します。結合に使う接続語句が与えられます。文章には終止符を1つしか使うことができないので忘れないようにしてください。大丈夫ですよ！"
    },
    "End: You Did it!": {
        "header": "よくできました！",
        "text": "これで終わりです！Quillクラス分けアクティビティを終了しました。アクティビティの結果は先生に送られ、学習計画の作成が始まります"
    },
    "completedDiagnostic": {
        "header": "Quillクラス分けアクティビティが完了しました",
        "text": "現在、結果を保存しています。保存が完了すると、自動的に移動します。"
    },
    "instructions": {
        "Fill in the blank with one of the choices.": "選んだ単語の1語を空欄に記入してください。",
        "Fill in each blank with one of the choices.": "各空欄に選んだ単語のうちの1語を記入してください。",
        "Fill in the blanks with one word each.": "半分終わりました！その調子で頑張りましょう",
        "Fill in the blanks with one word each. You can leave one empty.": "空欄に1語づつ単語を記入してください。",
        "Use the joining word to combine the sentences into one sentence.": "空欄に1語づつ単語を記入してください。１つの空欄は何も記入せずにそのままにすることができます"
    },
    "cues": {
        "choose one": "choose one / 1つ選ぶ",
        "choose three": "choose three / 3つ選ぶ",
        "choose two or three" : "choose two or three / 2つまたは3つ選ぶ",
        "joining word": "joining word / 単語の結合"
    },
    "feedback": {
        "blank": "下のオプションのうちの1つを使うか、空欄をそのままにします。",
        "no blank": "下のオプションのうちの1つを使います"
    }
}