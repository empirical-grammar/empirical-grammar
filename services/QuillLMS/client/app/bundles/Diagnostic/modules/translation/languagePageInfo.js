export const languages = ['english', 'arabic', 'chinese', 'french', 'hindi', 'spanish', 'vietnamese'];
export const languagesV2 = ['english', 'chinese', 'hindi', 'spanish', 'french', 'arabic', 'russian', 'portuguese', 'urdu', 'german', 'japanese', 'korean', 'vietnamese', 'thai', 'ukrainian', 'tagalog', 'dari'];
export const rightToLeftLanguages = ['arabic', 'urdu', 'dari'];
export const ENGLISH = 'english';

export const languageData = {
    'english': {
        flag: 'https://assets.quill.org/images/flags/usa.png',
        label: 'English'
    },
    'arabic': {
        flag: 'https://assets.quill.org/images/flags/egypt.png',
        label: 'العربية'
    },
    'chinese': {
        flag: 'https://assets.quill.org/images/flags/china.png',
        label: '中文'
    },
    'french': {
        flag: 'https://assets.quill.org/images/flags/france.png',
        label: 'Français'
    },
    'hindi': {
        flag: 'https://assets.quill.org/images/flags/india.png',
        label: 'हिंदी'
    },
    'spanish': {
        flag: 'https://assets.quill.org/images/flags/spain.png',
        label: 'Español'
    },
    'vietnamese': {
        flag: 'https://assets.quill.org/images/flags/vietnam.png',
        label: 'Tiếng Việt'
    }
}

export const languageDataV2 = {
    'english': {
        flag: 'https://assets.quill.org/images/flags/usa.png',
        label: 'English'
    },
    'chinese': {
        flag: 'https://assets.quill.org/images/flags/china.png',
        label: '中文'
    },
    'hindi': {
        flag: 'https://assets.quill.org/images/flags/india.png',
        label: 'हिंदी'
    },
    'spanish': {
        flag: 'https://assets.quill.org/images/flags/spain.png',
        label: 'Español'
    },
    'french': {
        flag: 'https://assets.quill.org/images/flags/france.png',
        label: 'Français'
    },
    'arabic': {
        flag: 'https://assets.quill.org/images/flags/egypt.png',
        label: 'العربية'
    },
    'russian': {
        flag: 'https://assets.quill.org/images/flags/russia.png',
        label: 'Русский'
    },
    'portuguese': {
        flag: 'https://assets.quill.org/images/flags/brazil.png',
        label: 'Português'
    },
    'urdu': {
        flag: 'https://assets.quill.org/images/flags/pakistan.png',
        label: 'اردو'
    },
    'german': {
        flag: 'https://assets.quill.org/images/flags/germany.png',
        label: 'Deutsche'
    },
    'japanese': {
        flag: 'https://assets.quill.org/images/flags/japan.png',
        label: '日本語'
    },
    'korean': {
        flag: 'https://assets.quill.org/images/flags/south_korea.png',
        label: '한국어'
    },
    'vietnamese': {
        flag: 'https://assets.quill.org/images/flags/vietnam.png',
        label: 'Tiếng Việt'
    },
    'thai': {
        flag: 'https://assets.quill.org/images/flags/thailand.png',
        label: 'ไทย'
    },
    'ukrainian': {
        flag: 'https://assets.quill.org/images/flags/ukraine.png',
        label: 'Українська'
    },
    'tagalog': {
        flag: 'https://assets.quill.org/images/flags/philippines.png',
        label: 'Tagalog'
    },
    'dari': {
        flag: 'https://assets.quill.org/images/flags/afghanistan.png',
        label: 'درى'
    },
}