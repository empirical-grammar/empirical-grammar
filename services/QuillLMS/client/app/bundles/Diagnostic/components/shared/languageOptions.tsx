import * as React from "react";

import { closeLanguageMenu, updateLanguage, } from '../../actions/diagnostics'
import { languages, languagesV2, languageData, languageDataV2 } from '../../modules/translation/languagePageInfo';
import i18n from '../../i18n';

interface LanguageOptionsProps {
  dispatch: (action: () => any) => void;
  diagnosticID: string;
}

export const LanguageOptions = ({ diagnosticID, dispatch, }: LanguageOptionsProps) => {
  // once we remove the original ELL Diagnostic, we can move to have only have the second versions
  let langs = diagnosticID === 'ell' ? languages : languagesV2;
  let langData = diagnosticID === 'ell' ? languageData : languageDataV2;

  const handleClickLanguage = (e) => {
    const language = e.currentTarget.value;
    i18n.changeLanguage(language);
    dispatch(updateLanguage(language));
    dispatch(closeLanguageMenu())
  }

  return langs.map(language => {
    return(
      <button className="language-button" key={`${language}-button`} onClick={handleClickLanguage} type="button" value={language}>
        <img alt={`${language} flag`} src={langData[language].flag} />
        <span>{langData[language].label}</span>
      </button>
    );
  })
}
