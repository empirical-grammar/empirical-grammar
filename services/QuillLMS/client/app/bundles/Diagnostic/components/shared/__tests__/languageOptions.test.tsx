import * as React from 'react';
import { shallow } from 'enzyme';

import { LanguageOptions } from '../languageOptions';
import { languages } from '../../../modules/translation/languagePageInfo';

const mockProps = {
  dispatch: () => jest.fn(),
  diagnosticID: 'ell-starter'
}

describe('LanguageOptions component', () => {
  it('should render correct languages for ELL Diagnostics', () => {
    const component = shallow(<LanguageOptions {...mockProps} />);
    expect(component).toMatchSnapshot();
    languages.forEach((language: string, i: number)=> {
      expect(component.find('button').at(i).props().value).toEqual(language);
    });
  });
})
