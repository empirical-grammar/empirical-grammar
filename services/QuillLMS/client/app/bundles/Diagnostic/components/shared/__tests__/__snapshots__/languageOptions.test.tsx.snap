// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageOptions component should render correct languages for ELL Diagnostic 1 1`] = `
Array [
  <button
    className="language-button"
    key="english-button"
    onClick={[Function]}
    type="button"
    value="english"
  >
    <img
      alt="english flag"
      src="https://assets.quill.org/images/flags/usa.png"
    />
    <span>
      English
    </span>
  </button>,
  <button
    className="language-button"
    key="chinese-button"
    onClick={[Function]}
    type="button"
    value="chinese"
  >
    <img
      alt="chinese flag"
      src="https://assets.quill.org/images/flags/china.png"
    />
    <span>
      中文
    </span>
  </button>,
  <button
    className="language-button"
    key="hindi-button"
    onClick={[Function]}
    type="button"
    value="hindi"
  >
    <img
      alt="hindi flag"
      src="https://assets.quill.org/images/flags/india.png"
    />
    <span>
      हिंदी
    </span>
  </button>,
  <button
    className="language-button"
    key="spanish-button"
    onClick={[Function]}
    type="button"
    value="spanish"
  >
    <img
      alt="spanish flag"
      src="https://assets.quill.org/images/flags/spain.png"
    />
    <span>
      Español
    </span>
  </button>,
  <button
    className="language-button"
    key="french-button"
    onClick={[Function]}
    type="button"
    value="french"
  >
    <img
      alt="french flag"
      src="https://assets.quill.org/images/flags/france.png"
    />
    <span>
      Français
    </span>
  </button>,
  <button
    className="language-button"
    key="arabic-button"
    onClick={[Function]}
    type="button"
    value="arabic"
  >
    <img
      alt="arabic flag"
      src="https://assets.quill.org/images/flags/egypt.png"
    />
    <span>
      العربية
    </span>
  </button>,
  <button
    className="language-button"
    key="russian-button"
    onClick={[Function]}
    type="button"
    value="russian"
  >
    <img
      alt="russian flag"
      src="https://assets.quill.org/images/flags/russia.png"
    />
    <span>
      Русский
    </span>
  </button>,
  <button
    className="language-button"
    key="portuguese-button"
    onClick={[Function]}
    type="button"
    value="portuguese"
  >
    <img
      alt="portuguese flag"
      src="https://assets.quill.org/images/flags/brazil.png"
    />
    <span>
      Português
    </span>
  </button>,
  <button
    className="language-button"
    key="urdu-button"
    onClick={[Function]}
    type="button"
    value="urdu"
  >
    <img
      alt="urdu flag"
      src="https://assets.quill.org/images/flags/pakistan.png"
    />
    <span>
      اردو
    </span>
  </button>,
  <button
    className="language-button"
    key="german-button"
    onClick={[Function]}
    type="button"
    value="german"
  >
    <img
      alt="german flag"
      src="https://assets.quill.org/images/flags/germany.png"
    />
    <span>
      Deutsche
    </span>
  </button>,
  <button
    className="language-button"
    key="japanese-button"
    onClick={[Function]}
    type="button"
    value="japanese"
  >
    <img
      alt="japanese flag"
      src="https://assets.quill.org/images/flags/japan.png"
    />
    <span>
      日本語
    </span>
  </button>,
  <button
    className="language-button"
    key="korean-button"
    onClick={[Function]}
    type="button"
    value="korean"
  >
    <img
      alt="korean flag"
      src="https://assets.quill.org/images/flags/south_korea.png"
    />
    <span>
      한국어
    </span>
  </button>,
  <button
    className="language-button"
    key="vietnamese-button"
    onClick={[Function]}
    type="button"
    value="vietnamese"
  >
    <img
      alt="vietnamese flag"
      src="https://assets.quill.org/images/flags/vietnam.png"
    />
    <span>
      Tiếng Việt
    </span>
  </button>,
  <button
    className="language-button"
    key="thai-button"
    onClick={[Function]}
    type="button"
    value="thai"
  >
    <img
      alt="thai flag"
      src="https://assets.quill.org/images/flags/thailand.png"
    />
    <span>
      ไทย
    </span>
  </button>,
  <button
    className="language-button"
    key="ukrainian-button"
    onClick={[Function]}
    type="button"
    value="ukrainian"
  >
    <img
      alt="ukrainian flag"
      src="https://assets.quill.org/images/flags/ukraine.png"
    />
    <span>
      Українська
    </span>
  </button>,
  <button
    className="language-button"
    key="tagalog-button"
    onClick={[Function]}
    type="button"
    value="tagalog"
  >
    <img
      alt="tagalog flag"
      src="https://assets.quill.org/images/flags/philippines.png"
    />
    <span>
      Tagalog
    </span>
  </button>,
  <button
    className="language-button"
    key="dari-button"
    onClick={[Function]}
    type="button"
    value="dari"
  >
    <img
      alt="dari flag"
      src="https://assets.quill.org/images/flags/afghanistan.png"
    />
    <span>
      درى
    </span>
  </button>,
]
`;

exports[`LanguageOptions component should render correct languages for ELL Diagnostic 2 1`] = `
Array [
  <button
    className="language-button"
    key="english-button"
    onClick={[Function]}
    type="button"
    value="english"
  >
    <img
      alt="english flag"
      src="https://assets.quill.org/images/flags/usa.png"
    />
    <span>
      English
    </span>
  </button>,
  <button
    className="language-button"
    key="arabic-button"
    onClick={[Function]}
    type="button"
    value="arabic"
  >
    <img
      alt="arabic flag"
      src="https://assets.quill.org/images/flags/egypt.png"
    />
    <span>
      العربية
    </span>
  </button>,
  <button
    className="language-button"
    key="chinese-button"
    onClick={[Function]}
    type="button"
    value="chinese"
  >
    <img
      alt="chinese flag"
      src="https://assets.quill.org/images/flags/china.png"
    />
    <span>
      中文
    </span>
  </button>,
  <button
    className="language-button"
    key="french-button"
    onClick={[Function]}
    type="button"
    value="french"
  >
    <img
      alt="french flag"
      src="https://assets.quill.org/images/flags/france.png"
    />
    <span>
      Français
    </span>
  </button>,
  <button
    className="language-button"
    key="hindi-button"
    onClick={[Function]}
    type="button"
    value="hindi"
  >
    <img
      alt="hindi flag"
      src="https://assets.quill.org/images/flags/india.png"
    />
    <span>
      हिंदी
    </span>
  </button>,
  <button
    className="language-button"
    key="spanish-button"
    onClick={[Function]}
    type="button"
    value="spanish"
  >
    <img
      alt="spanish flag"
      src="https://assets.quill.org/images/flags/spain.png"
    />
    <span>
      Español
    </span>
  </button>,
  <button
    className="language-button"
    key="vietnamese-button"
    onClick={[Function]}
    type="button"
    value="vietnamese"
  >
    <img
      alt="vietnamese flag"
      src="https://assets.quill.org/images/flags/vietnam.png"
    />
    <span>
      Tiếng Việt
    </span>
  </button>,
]
`;
